yaooqinn/kyuubi
# [KYUUBI #296] Add spark 2.4 profile and shade hive-sevice-rpc to engine jar to fix thrift build err

![yaooqinn](https://badgen.net/badge/Hello/yaooqinn/green) [![PR 296](https://badgen.net/badge/Preview/PR%20296/blue)](https://github.com/yaooqinn/kyuubi/pull/296) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)



### Please add issue ID here?

Fixes #296

### Why are the changes needed?


to make new Kyuubi support 2.4.7 for backward compatibility 

### Test Plan:
- Add some test cases that check the changes thoroughly including negative and positive cases if possible
- Add screenshots for manual tests if appropriate
- [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

This PR covers the thrift part build error and let us do the rest in followups
```logtalk
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetCatalogs.scala:37: value catalogManager is not a member of org.apache.spark.sql.internal.SessionState
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetColumns.scala:180: overloaded method value listTables with alternatives:
  (db: String,pattern: String)Seq[org.apache.spark.sql.catalyst.TableIdentifier] <and>
  (db: String)Seq[org.apache.spark.sql.catalyst.TableIdentifier]
 cannot be applied to (String, String, includeLocalTempViews: Boolean)
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetColumns.scala:181: value getTablesByName is not a member of org.apache.spark.sql.catalyst.catalog.SessionCatalog
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetColumns.scala:203: value listLocalTempViews is not a member of org.apache.spark.sql.catalyst.catalog.SessionCatalog
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetColumns.scala:206: value schema is not a member of Any
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetColumns.scala:208: value table is not a member of Any
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetColumns.scala:208: type mismatch;
 found   : Any
 required: org.apache.spark.sql.types.StructField
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetColumns.scala:208: type mismatch;
 found   : Any
 required: Int
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetTableTypes.scala:36: value tableTypes is not a member of object org.apache.spark.sql.catalyst.catalog.CatalogTableType
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetTables.scala:72: overloaded method value listTables with alternatives:
  (db: String,pattern: String)Seq[org.apache.spark.sql.catalyst.TableIdentifier] <and>
  (db: String)Seq[org.apache.spark.sql.catalyst.TableIdentifier]
 cannot be applied to (String, String, includeLocalTempViews: Boolean)
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetTables.scala:73: value getTablesByName is not a member of org.apache.spark.sql.catalyst.catalog.SessionCatalog
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetTables.scala:89: overloaded method value listTables with alternatives:
  (db: String,pattern: String)Seq[org.apache.spark.sql.catalyst.TableIdentifier] <and>
  (db: String)Seq[org.apache.spark.sql.catalyst.TableIdentifier]
 cannot be applied to (String, String, includeLocalTempViews: Boolean)
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetTables.scala:91: value listLocalTempViews is not a member of org.apache.spark.sql.catalyst.catalog.SessionCatalog
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/schema/RowSet.scala:27: object HiveResult is not a member of package org.apache.spark.sql.execution
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/schema/RowSet.scala:108: not found: value HiveResult
[ERROR] /Users/kentyao/kyuubi/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/schema/RowSet.scala:187: not found: value HiveResult
[ERROR] 16 errors found
```
